DTAdmin/Admins/NetworkManager.swift

```diff
@@ -0,0 +1,255 @@
+//
+//  NetworkManager.swift
+//  DTAdmin
+//
+//  Created by ITA student on 10/15/17.
+//  Copyright © 2017 Yurii Krupa. All rights reserved.
+//
+
+import Foundation
+
+class NetworkManager {
+  let urlProtocolPrefix = "http://"
+  let urlToHost = "vps9615.hyperhost.name"
+  let urlSuffixToUserLogIn = "/login/index"
+  let urlSuffixToUserLogOut = "/login/logout"
+  let urlSuffixToAdmins = "/AdminUser"
+  let urlSuffixToGetRecords = "/getRecords"
+  let urlSuffixToInsertData = "/insertData"
+  let urlSuffixToGetRecordById = "/getRecords"
+  let urlSuffixToUpdateRecord = "/update"
+  let urlSuffixToDeleteRecord = "/del"
+  //concat += </id> to update || delete exact record Exp route -> ./../entity/del/<id>
+  
+  private enum Credentials: String {
+    case userName = "username"
+    case password = "password"
+    case passwordConfirm = "password_confirm"
+  }
+  
+  private func cooikiesGetter(from hostUrl: String) -> [HTTPCookie] {
+    let cookies:[HTTPCookie] = HTTPCookieStorage.shared.cookies! as [HTTPCookie]
+    return cookies
+  }
+  private func requestBasic(with url: URL, method: String) -> URLRequest {
+    var request = URLRequest(url: url)
+    request.httpMethod = method
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    request.setValue("utf-8", forHTTPHeaderField: "Charset")
+    
+    return request
+  }
+  private func requestBasicWithBody(httpBody: Data, url: URL, method: String) -> URLRequest {
+    var request = requestBasic(with: url, method: method)
+    request.httpBody = httpBody
+    
+    return request
+  }
+  private func requestWithCookie(url: URL, method: String) -> URLRequest? {
+    if UserDefaults.standard.isLoggedIn(), let cookieValue = UserDefaults.standard.getCookie() {
+      var request = requestBasic(with: url, method: method)
+      request.setValue("session=\(cookieValue)", forHTTPHeaderField: "Cookie")
+      return request
+    }
+    return nil
+  }
+  private func requestWithCookie(and body: Data, url: URL, method: String) -> URLRequest? {
+    if var request = requestWithCookie(url: url, method: method) {
+      request.httpBody = body
+      return request
+    } else { return nil }
+  }
+  
+  func logIn(username: String, password: String, completionHandler: @escaping (_ user: UserModel.Admin, _ cookie: String) -> ()){
+    let credentials = [Credentials.userName.rawValue: username, Credentials.password.rawValue: password]
+    guard let httpBody = try? JSONSerialization.data(withJSONObject: credentials, options: []) else {return}
```

Comment:
My concern here is that in case JSONSerialization returns error we will not know about it. It's a good case to use preconditionFailure("JSON serialization failed") (See details for others at http://blog.krzyzanowskim.com/2015/03/09/swift-asserts-the-missing-manual/)